specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,261 @@
+    - [Chain ID Frontrunning](#chain-id-frontrunning)
+    - [Chain ID Value](#chain-id-value)
+    - [Proxy Admin Owner](#proxy-admin-owner)
+    - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deployment
+
+The OP Stack Manager is deployed at `0xTODO`. It can be deployed as follows:
+
+TODO.
+
+## Interface
+
+Version 1.0.0 of the OP Stack Manager deploys the [`op-contracts/v1.3.0`]
+contracts release.
+
+### `deploy`
+
+The `deploy` method is the only non-view method in the contract. It is used to
+deploy the full set of L1 contracts required to setup a new OP Stack chain that
+complies with the [standard configuration]. It has the following interface:
+
+```solidity
+function deploy(
+  uint64 l2ChainId,
+  address proxyAdminOwner,
+  SystemConfigInputs memory systemConfigInputs,
+  L2OutputOracleInputs memory l2OutputOracleInputs
+) external returns (SystemConfig systemConfig);
+```
+
+where:
+
+- `l2ChainId` is the chain ID of the OP Stack chain.
+- `proxyAdminOwner` is the address that will own the `ProxyAdmin` contract that will be deployed.
+This corresponds to the L1 ProxyAdmin owner and L2 ProxyAdmin owner [admin roles].
+- `systemConfigInputs` is a struct defining configuration of the `SystemConfig` inputs.
+- `l2OutputOracleInputs` is a struct defining configuration of the `L2OutputOracle` inputs.
+
+The `l2ChainId` has the following restrictions:
+
+- It is limited to a `uint64`. This contradicts the [interop specification],
+which allows chain ID to be a `uint256`. This contradiction exists because it is
+currently unspecified how to unify a 256-bit chain ID with the existing batch inbox
+address convention, as tracked in [ethereum-optimism/specs#94]. This limitation
+may be removed in a future version of the OP Stack Manager once a solution to
+that issue is found.
+- It must not be equal to 0.
+- It must not be equal to the chain ID of the chain the OP Stack Manager is
+deployed on.
+- It must not be equal to a chain ID that is already present in the
+[ethereum-lists/chains] repository. This is not enforced onchain, but may matter
+for future versions of OP Stack Manager that handle upgrades via a privileged role.
+
+The `SystemConfigInputs` struct is defined as follows:
+
+```solidity
+struct SystemConfigInputs {
+  address systemConfigOwner;
+  uint256 overhead;
+  uint256 scalar;
+  bytes32 batcherHash;
+  address unsafeBlockSigner;
+}
+```
+
+Details on each parameter can be found in the code comments of the [`SystemConfig` contract].
+Note that not all `SystemConfig` parameters are able to be configured.
+This is because the OP Stack Manager enforces compliance with the [standard configuration], and only allows
+configuration where allowed by the standard configuration.
+
+The `L2OutputOracleInputs` struct is defined as follows:
+
+```solidity
+struct L2OutputOracleInputs {
+  address proposer;
+  address challenger;
+}
```

Comment:
These roles still exist post fault-proofs (for the permissioned game) and aren't tied to the L2OutputOracle so possibly we should just rename this struct to be specifically about defining system roles.

Reply:
Related to https://github.com/ethereum-optimism/specs/pull/236#discussion_r1642911392, I like this idea of a `Roles` struct to help minimize the changes between versions. I'll make this change

Reply:
Added in 9633e2e